extensions/standard-processors/tests/unit/PutTCPTests.cpp

```diff
@@ -400,9 +403,9 @@ TEST_CASE("PutTCP test missing client cert", "[PutTCP]") {
   auto port = test_fixture.addSSLServer();
   test_fixture.setPutTCPPort(port);
 
-  trigger_expect_failure(test_fixture, "message for invalid-cert server");
+  test_fixture.trigger("message for invalid-cert server");
```

Comment:
I know this is due to a change in TLS (you mentioned in the PR aswell)
`Fix tests to not expect error failed handshake in case of missing certificate as of TLS 1.3 handshake failure is only detected by the client on the first read operation: https://stackoverflow.com/questions/62459802/tls-1-3-client-does-not-report-failed-handshake-when-client-certificate-verifica`

But I think we should emphasize the meaning of this:
If someone has an invalid client certification on a minifi client which is not accepted by the server (the message will be disregarded), then using TLSV1.2 will route the messages to `failure` however using TLSv1.3 can (most often than not based on my testing) route these to `success` (sometimes `failure`)

This means feature loss on v1.3 compared to v1.2.
Do you see any way how we can resolve this?
If not then I think we should thoroughly document this phenomenon in PROCESSORS.md

Reply:
When I tested it, it seemed the routing depended on if the message could be sent before the server reset the connection. If it was reset before the message was sent it resulted in failure otherwise the result was success. Unfortunately in this processor we do not read through the connection, only write to the socket, so we cannot know if the handshake succeeded or not using TLS 1.3 other than the connection was reset or not while sending the message. We can document this, but I do not see any other option how this could be handled.

Reply:
The failure relationship is documented like this at the moment: "FlowFiles that encountered IO errors are send out this relationship."

It does not say anything about the message arriving on the server or not, and I think no IO errors are reported if the handshake fails so it can match the current functionality, but I suppose it's better to add a note about this.

Reply:
Added a note in 195fe306286c57fdb91e00c62c014172ccd6c8b4

Reply:
Yeah thats what I feared, without a more complex application layer we cant gurantee anything.

Still a note could be useful to the user, but I am not sure where can we put something like this.
The processor description seems a bit generic for a note like this, but anything other than that will be overwritten by the doc generation. 🤷 Maybe a code comment wil be sufficient?